Move phpunit-polyfills to dev-requirements

heiglandreas opened the pull request:
Without PHPUnit these polyfills do not make any sense. In addition to that not having them in the dev-dependencies will download phpunit  into any project regardless of whether the project uses phpunit or not.

I would even argue that `roots/wordpress` should be a dev- requirement as using this package with any other composer based wordpress installation will download wordpress twice - and possibly overwrite the installation with a different version resulting in unpredictable results – which is not really what one wants in a testing setup

heiglandreas committed:
Move phpunit-polyfills to dev-requirements

Without PHPUnit these polyfills do not make any sense. In addition to that not having them in the dev-dependencies will download phpunit  into any project regardless of whether the project uses phpunit or not.

I would even argue that ´roots/wordpress` should be a dev- requirement as using this package with any other composer based wordpress installation will download wordpress twice - and possibly overwrite the installation with a different version resulting in unpredictable results – which is not really what one wants in a testing setup

heiglandreas committed:
Create wordpress dir when missing

When this package is installed before the wordpress-package, the wordpress folder is missing which will lead to a failure in moving the config-file.

This change fixes that by creating the wordpress folder should it not exist already

szepeviktor commented on src/Composer/InstallDropin.php:
```diff
@@ -9,7 +9,10 @@
+			mkdir( 'wordpress' );
+		}
+		if ( ! is_dir(  'wordpress/wp-content' ) ) {
 			mkdir( 'wordpress/wp-content' );
```
Thank you for the PR.

You can simply
```php
mkdir( 'wordpress/wp-content', 0777, true );
```

https://www.php.net/manual/en/function.mkdir.php

szepeviktor committed:
Upgrade lock file

heiglandreas committed:
Make changes less intrusive

Instead of adding additional code it is easier to use the recursive-feature of `mkdir`.

I didn't use named arguments as that would require PHP8.0 but the code should be – at least for now – usable with lower versions as well

leogermani commented:
> I would even argue that roots/wordpress should be a dev- requirement 

WorDBless is usually a dev requirement of another project, but wordpress core will be needed there, so it can't be a dev dependency of WorDBless... but there's an open discussion to move the files somewhere else -> https://github.com/Automattic/wordbless/issues/32

heiglandreas commented:
Best option would actually be to require **a** WordPress. But that would mean that there is a meta-package "wordpress-implementation" that this could then require.

So this one would

```json
{
    "require": {
        "wordpress-implementation": "^6.4"
    }
}
```

and automatic/wordpress and f.e. johnpblock/wordpress would both contain 

```json
{
    "provide": {
        "wordpress-implementation": "6.4"
    }
}
```

But as the later isn't available currently.... :shrug:
